samples/pollion_dash/data_preparation.R

```diff
@@ -0,0 +1,178 @@
+
+
+##create vector of packages to either install or load
+packages_to_install <- c("ggplot2","shiny","reshape","scales","grid","gridExtra",
+                         "rjson","tm","wordcloud","RColorBrewer","psych")
+packages_to_install
+
+#install.packages(packages_to_install)
+##use lapply to be able to use library() despite quatation marks
+lapply(packages_to_install, library, character.only=T)
+
+############################ data manipulation
+#get and set working directory
+working_dir<-getwd()
+working_dir
+
+startpoll_json_file<-"/Users/Thomas/Documents/app_pollion/app_pasta_neu/data/pasta_data2.json"
+
+
+#load data
+poll_data<-fromJSON(file=startpoll_json_file)
+
+#in case of empty list element, replace by NA
+poll_data <- lapply(poll_data, function(x) {
+        as.data.frame(replace(x, sapply(x, is.list), NA))
+})
+
+#use rbind.fill to automatically fill empty elements to avoid messing up number of columns
+poll_data <- rbind.fill(poll_data)
+poll_data
+head(poll_data)
+names(poll_data)
+table(poll_data$citySize)
+
+oldnames<-c(colnames(poll_data))
+oldnames
+
+newnames<-c("X_v","X_id",                  
+            "creationtime","finished",
+            "pollid","publicpoll",
+            "userid","gender",
+            "birthYear","citySize",
+            "organicConsumptionFrequency1","pasta_sorte",           
+            "erwartung_erfullt","bewertung_geschmack",          
+            "bewertung_konsistenz","wieder_kaufen",           
+            "weiter_empfehlen","gut_offen",          
+            "schlecht_offen","gender",          
+            "birthYear","organicConsumptionFrequency")
+
+for(i in 1:length(colnames(poll_data))) {
+        names(poll_data)[names(poll_data)==oldnames[i]]=newnames[i]
+}
+
+head(poll_data)
+
+data<-poll_data[,c(8:19)]
+
+names(data)
+
+#gender
+level_reorder_gender <- c("female","male","not answered")
+level_labels_gender <- c("weiblich","mannlich","keine Angabe")
+
+func_categorial("gender",level_reorder_gender,level_labels_gender)
+
+#year of birth
+upper_bound_birthyear <- 1990
+lower_bound_birthyear <- 1900
+
+func_integer("birthYear",upper_bound_birthyear,lower_bound_birthyear)
+
+#city of residence
+levels(data$citySize)
+level_reorder_city <- c("less than 5000","5000 - 19999","20000 - 99999","100000 - 1000000","more than 1000000","not answered")
+level_labels_city <- c("weniger als 5000","5000 - 19999","20000 - 99999","100000 - 1000000","mehr als 1000000","keine Angabe")
+
+func_ordinal("citySize",level_reorder_city,level_labels_city)
+
+##################make sure how to deal with socio demographic dublicates!!!!!!!!!!!!
+
+#organicConsumptionFrequency
+level_reorder_consumption=c("daily","weekly","monthly")
+level_label_consumptionlabels=c("taglich","wochentlich","monatlich")
+
+func_ordinal("organicConsumptionFrequency1",level_reorder_consumption,level_label_consumptionlabels)
+
+
+#Pasta Sort
+level_reorder_pasta_sorte=c(levels(data$pasta_sorte)[1],levels(data$pasta_sorte)[4],
```

Comment:
variable names always and consistently only in english, we could have a programmer in our team who does not speak german